### PR TITLE
[tool] Update cts converter tool to support `android-cts-10.0_r2`

### LINKED ISSUE #1079: [test][tool] Upgrade CTS align with updated android-cts-10.0_rc1
- Try to use [cts/tool/CTSConverter/src/test_generator.py](https://github.com/intel/webml-polyfill/blob/master/test/cts/tool/CTSConverter/src/test_generator.py) to convert [last spec of android-cts-10.0_rc1]( https://android.googlesource.com/platform/frameworks/ml/+/master/nn/runtime/test/specs), find the gap

- Investigate the latest Android CTS convert script

- Relaxed test case align Android CTS Criteria

### PR DESCRIPTION
**Pull request by cuiyanx**
Update cts converter tool to support [nn (tag: android-cts-10.0_r2)](https://android.googlesource.com/platform/frameworks/ml/+/refs/tags/android-cts-10.0_r2).

fixes #1079

Signed-off-by: cuiyanx <yanx.cui@intel.com>

### CODE REVIEW

**Comment by cuiyanx** [comment]
@BruceDai PTAL

**Comment by cuiyanx** [comment]
@BruceDai PTAL

File: test/cts/tool/CTSConverter/README.md [hunk 1]
```diff
@@ -1,48 +1,57 @@
...
 
-### create `cts-all.js` into `./output`, this file include nn and plus test case
+Transfer nn test case into `./output/cts`
 
 ```shell
 $ npm start
 ```
 
-### get more information for command line
+## Update or create combine test file
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/README.md]: @cuiyanx Please modify ‘Update or create combine test file’ to be ‘Update or create combined test files’.

File: test/cts/tool/CTSConverter/src/main.py [hunk 2]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
-    support_cts_file = "./slice.txt"
+    output_path_transfer = os.path.join(output_path_transfer_clear, "specs")
+    support_cts_file = "./slice.json"
 
     with open(support_cts_file) as cts_file:
-      cts_file_names = cts_file.readlines()
+      cts_file_names = json.load(cts_file)
       print ("Open support cts file: " + support_cts_file + "\n")
 
     print ("transfer nn test case to js test case....\n")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: @cuiyanx Plase capitalize the first letter of  **transfer**.

File: test/cts/tool/CTSConverter/src/main.py [hunk 3]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
     transfer(args_transfer, output_path_transfer, cts_file_names)
 
   if not args_cts == "-":
-    print ("scan test cts directory....\n")
+    print ("scan test cts directory....")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Ditto

File: test/cts/tool/CTSConverter/src/main.py [hunk 4]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
     cts_file_dict = get_file_names(args_cts, ".js")
     file_dict_all.update(cts_file_dict)
 
   if not args_supplement == "-":
-    print ("scan test supplement directory....\n")
+    print ("scan test supplement directory....")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Ditto

File: test/cts/tool/CTSConverter/src/main.py [hunk 5]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
 
     describeString = "CTS Supplement Test"
 
     supplement_file_dict = get_file_names(args_supplement, ".js")
     file_dict_all.update(supplement_file_dict)
 
   if not args_plus == "-":
-    print ("scan test plus directory....\n")
+    print ("scan test plus directory....")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Ditto

File: test/cts/tool/CTSConverter/src/main.py [hunk 6]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
+    skip_files = ["cts-all.js", "cts_supplement-all.js", "conv_65_65_96.js"]
+    (args_all_path, args_all_name) = os.path.split(args_all)
+    skip_files.append(args_all_name)
+
+    for name in skip_files:
+      if name in file_dict_all:
+        print ("skip creating file name: %s"%file_dict_all[name])
+        del file_dict_all[name]
+
+    print ("reordering by name....")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Ditto

File: test/cts/tool/CTSConverter/src/main.py [hunk 7]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
     file_list = sorted(file_dict_all.keys())
 
-    print ("create all test case file....\n")
+    print ("create all test case file: %s"%str(output_file_all))
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Ditto
And modify **all test case file** to **all-in-one test case file**.

File: test/cts/tool/CTSConverter/src/main.py [hunk 8]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
     create(output_file_all, file_dict_all, file_list, describeString)
 
-  print ("transfer and create all files are completed")
+  if not args_transfer == "-":
+    if not args_all == "-":
+      print ("transfer and create all files are completed")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Please modify **transfer and create all files are completed** to **Completed to transfer Android CTS Spec files.**

File: test/cts/tool/CTSConverter/src/main.py [hunk 9]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
+    else :
+      print ("transfer all files are completed")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Ditto

File: test/cts/tool/CTSConverter/src/main.py [hunk 10]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
+  else :
+    if not args_all == "-":
+      print ("create all files are completed")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Ditto

File: test/cts/tool/CTSConverter/src/main.py [hunk 11]
```diff
@@ -97,55 +97,79 @@ def create(opath, file_dict, file_list, describe):
...
+    else :
+      print ("nothing to do!! get info with 'npm run info' command")
```
Comment (BruceDai) [inline test/cts/tool/CTSConverter/src/main.py]: Ditto

**Comment by cuiyanx** [comment]
@BruceDai PTAL

**Comment by BruceDai** [comment]
LGTM, thanks @cuiyanx 
@ibelem Please help merge this PR, thanks.

**Comment by ibelem** [comment]
Thanks for the great efforts @cuiyanx !

**Pull request merged**